theseus/core/objective.py

```diff
@@ -107,6 +107,27 @@ def _get_cost_function_variable_batch_name(var: Manifold):
             + f"{tuple(var.data.shape[1:])}"
         )
 
+    @staticmethod
+    def _get_cost_function_batch_name(cost_function: CostFunction) -> Union[str, None]:
```

Comment:
@luisenp @mhmukadam You don't have to review this PR.  If you have time, please check if this function generates unique batch name for cost functions under the assumption that all the `optim_vars` and `aux_vars` have the same batch size.

Reply:
LGTM!